auth/utils/read-external-sources.ts

```diff
@@ -96,9 +96,50 @@ function filterPropertiesByServiceName(envObj: any, serviceName: string): any {
 /**
  * Pulls credentials from VCAP_SERVICES env property that IBM Cloud sets
  *
+ * The function will first look for a service entry whose "name" field matches
+ * the serviceKey value. If found, return its credentials.
+ *
+ * If no match against the service entry's "name" field is found, then find the
+ * service list with a key matching the serviceKey value. If found, return its
+ * credentials.
```

Comment:
Should mention that you are returning the first credentials object in the service list.